src/library/simplified/extension/marketingABTests.js

```diff
@@ -0,0 +1,17 @@
+var marketingABTests = "3659,14698,14739,13542,12102";
+var marketingABTestsArray = marketingABTests.split(",");
+var abTests = utag_data.context.tests;
+var marketingABTests = "";
+
+for (var abTest=0; abTest<abTests.length; abTest++) {
+    for (var marketingABTest=0; marketingABTest<marketingABTestsArray.length; marketingABTest++) {
```

Comment:
We just need a single for loop to flatten the "tests" list, where you grab all the tests and their corresponding buckets and combined them all into one long string. Basically all you need to do for this extension is iterate through "tests" list and take each test and its bucket to concat them into the string format "testId.bucketId|testId.bucketId|testId.bucketId|...", and assign the string to a new udo variable, in this case marketingABTests, that you create in this extension. That's pretty much it, just make sure you set this extension to run prior to load rules. Then this string will be used in a load rule. For example, if we want a load rule to fire a tag based on a given marketing test ID defined (or given by) marketing team, let's say 12345, we will then have a load rule to check and see if "12345.1" exists in the string, marketingABTests, created by this extension. If the "marketingABTests" contains "12345.1".  I have outlined the steps for this under "Tealium" section of the confluence page.